collateral.md

```diff
@@ -233,5 +240,52 @@ module COLLATERAL
          ...</flip-bids>
       requires GUY =/=K 0 andBool BID <Rat TAB
 
+    syntax MCDContract ::= GemJoinContract
+    syntax GemJoinContract ::= "GemJoin" String
+    syntax MCDStep ::= GemJoinContract "." GemJoinStep [klabel(gemJoinStep)]
+ // ------------------------------------------------------------------------
+    rule contract(GemJoin GEMID . _) => GemJoin GEMID
+    rule [[ address(GemJoin GEMID) => ACCTJOIN ]] <gemJoin-gem> GEMID </gemJoin-gem> <gemJoin-addr> ACCTJOIN </gemJoin-addr>
+    rule <k> GemJoin _ . _ => exception ... </k> [owise]
+
+    syntax GemJoinStep ::= "join" Address Wad
+ // -----------------------------------------
+    rule <k> GemJoin GEMID . join USR AMOUNT
+          => call Vat . slip GEMID USR AMOUNT
+          ~> call Gem GEMID . transferFrom MSGSENDER THIS AMOUNT ... </k>
+         <msg-sender> MSGSENDER </msg-sender>
+         <this> THIS </this>
```

Comment:
I think we should check that `AMOUNT` is positive. The solidity code effectively does, and it's not clear to me that a negative AMOUNT here makes much sense (esp. since any sensible implementation of `transferFrom` wouldn't allow it).